examples/usb_i2c_mctp.rs

```diff
@@ -0,0 +1,225 @@
+/// Debug LED config, change these based on the board config
+pub const PANIC_LED: u32 = 0;
+/// Triggered when waiting for RX
+pub const RX_LED: u32 = 1;
+/// Triggered when TX in progress
+pub const TX_LED: u32 = 2;
+
+/// # Summary
+///
+/// A helper function to append the packet metadata do the outgoing buffer
+/// pointed to by @buf.
+///
+/// # Parameter
+///
+/// * `buf`: the buffer in which to append the metadata to.
+/// * `n_bytes`: length of data message body of the next message this is
+///              stored in the packet header, so that the host can
+///              determine the data message read length.
+///
+/// # Returns
+///
+/// Ok(n) returns the total size of bytes to send from this buffer (buf[0..n]).
+///
+/// # Panics
+///
+/// If the buffer capacity cannot fit the packet metadata
+///
+fn prepare_tx_header(buf: &mut [u8], n_bytes: usize) -> Result<usize, ()> {
+    let total_bytes_to_send = HEADER_LEN;
+    if total_bytes_to_send > buf.len() {
+        return Err(());
+    }
+    // Setup Header
+    buf[0] = 0xBB;
+    buf[1] = 0xFF;
+    // Upper 8-bits
+    buf[2..=3].copy_from_slice(&u16::to_be_bytes(n_bytes as u16));
+
+    Ok(total_bytes_to_send)
+}
+
+fn main() {
+    let led_count = Leds::count().unwrap_or(0);
+
+    // Using the led number, set it on iff it's available
+    let led_on = |led_num| {
+        if led_num < led_count {
+            Leds::on(led_num).unwrap()
+        }
+    };
+
+    // Using the led number, set it off iff it's available
+    let led_off = |led_num| {
+        if led_num < led_count {
+            Leds::off(led_num).unwrap()
+        }
+    };
+
+    // RX Buffer layout
+    // [0] = Preamble
+    // [1] = TargetID
+    // [2] = Length Upper Byte
+    // [3] = Length Lower Byte
+    let mut rx_buf: [u8; RX_BUF_LEN] = [0x00; RX_BUF_LEN];
+    let mut msg_len: u16;
+    let mut target_id: u16;
+
+    loop {
+        led_on(RX_LED);
+        let (_, err) = Console::read(&mut rx_buf);
+        led_off(RX_LED);
+
+        if err.is_err() {
+            led_on(PANIC_LED);
+            panic!("Failed to read from host {:?}", err);
+        }
+        // If we don't get a matching preamble, then the rest of the data is unreliable.
+        assert_eq!(rx_buf[0], 0xAA);
+        // Target in 7-bit address range?
+        assert!(rx_buf[1] <= 0x7F);
+        target_id = rx_buf[1] as u16;
+        // Data length should be non-zero, otherwise why are we here? just to suffer?
+        msg_len = u16::from_be_bytes([rx_buf[2], rx_buf[3]]);
+        assert!(msg_len as usize <= MAX_DLEN);
+        assert_ne!(msg_len, 0);
+
+        if let Err(why) = I2CMasterSlave::i2c_master_slave_write_sync(
+            target_id as u16,
+            &mut rx_buf[HEADER_LEN..HEADER_LEN + msg_len as usize],
+            msg_len as u16,
+        ) {
+            led_on(PANIC_LED);
+            panic!("i2c-master: write operation failed {:?}", why);
+        } else {
```

Comment:
Also we can drop this else

Reply:
fixed!

Reply:
I removed some of the other `else` clauses as well since we panic on the `if`... little bit easier to read.